geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/DestroyMappingCommandDunitTest.java

```diff
@@ -171,11 +203,80 @@ public void destroysSynchronousMapping() throws Exception {
+      verifyQueueRemoved(cache, null, false);
+    });
+  }
+
+  @Test
+  public void destroysMappingForServerGroup() throws Exception {
+    setupMappingWithServerGroup(TEST_GROUP1, GROUP1_REGION, true);
+    CommandStringBuilder csb =
+        new CommandStringBuilder(DESTROY_MAPPING + " --groups=" + TEST_GROUP1);
+    csb.addOption(REGION_NAME, GROUP1_REGION);
+
+    gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess();
+
+    locator.invoke(() -> {
+      assertThat(getRegionMappingFromClusterConfig()).isNull();
+      validateAsyncEventQueueRemovedFromClusterConfig();
+      validateRegionAlteredInClusterConfig(true);
+    });
+
+    // we have this region on server2 only
+    server2.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      verifyMappingRemovedFromService(cache, GROUP1_REGION, false);
+      verifyRegionAltered(cache, GROUP1_REGION, false);
+      verifyQueueRemoved(cache, GROUP1_REGION, false);
+    });
+  }
+
+  @Test
+  public void destroysMappingForMultiServerGroup() throws Exception {
+    setupMappingWithServerGroup(TEST_GROUP1 + "," + TEST_GROUP2, GROUP1_GROUP2_REGION, true);
+    // Purposely destroy the mapping on one group only
+    CommandStringBuilder csb =
+        new CommandStringBuilder(DESTROY_MAPPING + " --groups=" + TEST_GROUP1);
+    csb.addOption(REGION_NAME, GROUP1_GROUP2_REGION);
+
+    gfsh.executeAndAssertThat(csb.toString()).statusIsSuccess();
+
+    locator.invoke(() -> {
+      assertThat(getRegionMappingFromClusterConfig()).isNull();
+      validateAsyncEventQueueRemovedFromClusterConfig();
+      validateRegionAlteredInClusterConfig(true);
+    });
+
+    // server1 never has the mapping
+    server1.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      verifyMappingRemovedFromService(cache, GROUP1_GROUP2_REGION, false);
+      verifyRegionAltered(cache, GROUP1_GROUP2_REGION, false);
+      verifyQueueRemoved(cache, GROUP1_GROUP2_REGION, false);
+    });
+    // server2 and server4's mapping are destroyed
+    server2.invoke(() -> {
       InternalCache cache = ClusterStartupRule.getCache();
-      verifyMappingRemovedFromService(cache);
-      verifyRegionAltered(cache);
-      verifyQueueRemoved(cache);
+      verifyMappingRemovedFromService(cache, GROUP1_GROUP2_REGION, false);
+      verifyRegionAltered(cache, GROUP1_GROUP2_REGION, false);
+      verifyQueueRemoved(cache, GROUP1_GROUP2_REGION, false);
+    });
+    server4.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      verifyMappingRemovedFromService(cache, GROUP1_GROUP2_REGION, false);
+      verifyRegionAltered(cache, GROUP1_GROUP2_REGION, false);
+      verifyQueueRemoved(cache, GROUP1_GROUP2_REGION, false);
+    });
+    // server3 should still have the mapping
+    server3.invoke(() -> {
+      InternalCache cache = ClusterStartupRule.getCache();
+      verifyMappingRemovedFromService(cache, GROUP1_GROUP2_REGION, true);
```

Comment:
Now this method name `verifyMappingRemovedFromService ` is misleading. Actually we want to verify the mapping is *not* removed.

Can we change the return value from `void` to `boolean` and rename the method a little bit? So that we can use `assertThat` and make the code easier to read and understand. e.g. for this case, it will be `assertThat(mappingRemovedFromService).isFalse()`

Reply:
✅